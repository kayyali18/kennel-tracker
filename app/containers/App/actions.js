/*
 * App Actions
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 */

import {
  LOAD_API,
  LOAD_API_SUCCESS,
  LOAD_API_ERROR,
  LOGIN_SUCCESS,
  LOGIN_ERROR,
} from './constants'

/**
 * Load the api data, this action starts the request saga
 *
 * @return {object} An action object with a type of LOAD_API
 */
export function loadApi() {
  return {
    type: LOAD_API,
  }
}

/**
 * Dispatched when the API suggestions are loaded by the request saga
 *
 * @param  {object} results The api response data
 *
 * @return {object} An action object with a type of LOAD_API_SUCCESS passing the result
 */
export function apiLoaded(results) {
  return {
    type: LOAD_API_SUCCESS,
    results,
  }
}

/**
 * Dispatched when loading the API fails
 *
 * @param  {object} error The error
 *
 * @return {object} An action object with a type of LOAD_API_ERROR passing the error
 */

export function apiLoadingError(error) {
  return {
    type: LOAD_API_ERROR,
    error,
  }
}

/**
 * Dispatched when login credentials are credible
 *
 * @param  {object} response The response with expiration date and bearer token
 *
 * @return {object} An action object with a type of LOGIN_SUCCESS passing the response
 */

export function loginSuccess(response) {
  return {
    type: LOGIN_SUCCESS,
    response,
  }
}

/**
 * Dispatched when login credentials are uncredible
 *
 * @param  {object} error The response error
 *
 * @return {string} An action object with a type of LOGIN_ERROR passing the error
 */

export function loginError(error) {
  return {
    type: LOGIN_ERROR,
    error,
  }
}
