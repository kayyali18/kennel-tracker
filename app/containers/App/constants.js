/*
 * AppConstants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */

export const LOAD_API = 'kennel-tracker/App/LOAD_API'
export const LOAD_API_SUCCESS = 'kennel-tracker/App/LOAD_API_SUCCESS'
export const LOAD_API_ERROR = 'kennel-tracker/App/LOAD_API_ERROR'
export const LOGIN_SUCCESS = 'kennel-tracker/App/LOGIN_SUCCESS'
export const LOGIN_ERROR = 'kennel-tracker/App/LOGIN_ERROR'
