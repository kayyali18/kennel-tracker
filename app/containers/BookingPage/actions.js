/*
 * Booking Actions
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 */

import {
  POST_DOG,
  POST_OWNER,
  POST_VET,
  POST_SUCCESS,
  POST_ERROR,
  RUN_POST_SAGA,
  SET_PATH,
} from './constants'

/**
 * This actions starts the POST request saga watcher
 *
 * @return {object} An action object with a type of POST_SAGA
 */
export function runPostSagaWatcher() {
  return {
    type: RUN_POST_SAGA,
  }
}

/**
 * Dispatched when form is submitted, updates owner information in store
 *
 * @param  {object} body The booking form entries for owner
 *
 * @return {object} An action object with a type of POST_OWNER passing the body
 */
export function postOwner(body) {
  return {
    type: POST_OWNER,
    body,
  }
}

/**
 * Dispatched when form is submitted, updates pet information in store
 *
 * @param  {object} body The booking form entries for pet
 *
 * @return {object} An action object with a type of POST_DOG passing the body
 */
export function postDog(body) {
  return {
    type: POST_DOG,
    body,
  }
}

/**
 * Dispatched when form is submitted, updates vet information in store
 *
 * @param  {object} body The booking form entries for the vet
 *
 * @return {object} An action object with a type of POST_VET passing the body
 */
export function postVet(body) {
  return {
    type: POST_VET,
    body,
  }
}

/**
 * Dispatched on form submit to set path for saga to fetch
 *
 * @param  {string} path The path for the saga to POST to
 *
 * @return {object} An action object with a type of POST_ERROR passing the path
 */

export function setPath(path) {
  return {
    type: SET_PATH,
    path,
  }
}

/**
 * Dispatched when POST req to the server fails
 *
 * @param  {object} error The error
 *
 * @return {object} An action object with a type of POST_ERROR passing the error
 */

export function postError(error) {
  return {
    type: POST_ERROR,
    error,
  }
}

/**
 * Dispatched when POST is successfull
 *
 * @param {object} response A response object indicating successfull POST
 *
 * @return {object} An action object with a type of POST_SUCCESS setting authentication to true
 */

export function postSuccess(response) {
  return {
    type: POST_SUCCESS,
    message: response.message,
  }
}
