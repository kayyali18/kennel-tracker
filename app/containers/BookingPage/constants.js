/*
 * BookingPage Constants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */

export const POST_OWNER = 'kennel-tracker/App/POST_OWNER'
export const POST_DOG = 'kennel-tracker/App/POST_DOG'
export const POST_VET = 'kennel-tracker/App/POST_VET'
export const RUN_POST_SAGA = 'kennel-tracker/App/RUN_POST_SAGA'
export const POST_SUCCESS = 'kennel-tracker/App/POSTSUCCESS'
export const POST_ERROR = 'kennel-tracker/App/POST_ERROR'
export const SET_PATH = 'kennel-tracker/App/SET_PATH'
