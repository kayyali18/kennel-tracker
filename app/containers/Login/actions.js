/*
 * Login Actions
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 */

import {
  UPDATE_TOKEN,
  RUN_TOKEN_SAGA,
  SUBMIT_USER_CREDS,
  CLEAR_USER_CREDS,
} from './constants'

/**
 * Dispatched when login credentials are submitted
 *
 * @param  {object} user The user with object containing email and pass
 *
 * @return {object} An action object with a type of SUBMIT_USER_CREDS passing the response
 */

export function submitUserCredentials(user) {
  return {
    type: SUBMIT_USER_CREDS,
    user,
  }
}

/**
 * Dispatched when login credentials are succesfull
 *
 *
 * @return {object} An action object with a type of CLEAR_USER_CREDS passing the response
 */

export function clearUserCredentials() {
  return {
    type: CLEAR_USER_CREDS,
  }
}

/**
 * Dispatched when login credentials are submitted
 *
 * @param  {object} response The response with expiration date and bearer token
 *
 * @return {object} An action object with a type of UPDATE_TOKEN passing the response
 */

export function updateToken(response) {
  return {
    type: UPDATE_TOKEN,
    response,
  }
}

/**
 * Fires off saga watch for this action type
 */
export function runTokenSagaWatcher() {
  return {
    type: RUN_TOKEN_SAGA,
  }
}
