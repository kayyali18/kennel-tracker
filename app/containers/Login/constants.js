/*
 * Login Constants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */

export const UPDATE_TOKEN = 'kennel-tracker/Home/UPDATE_TOKEN'
export const RUN_TOKEN_SAGA = 'kennel-tracker/Home/RUN_TOKEN_SAGA'
