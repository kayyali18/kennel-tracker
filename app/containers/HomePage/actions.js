/*
 * Home Actions
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 */

import { DISPLAY_CURR_DOGS, RUN_CURR_DOGS_SAGA } from './constants'

/**
 * Changes the input field of the form
 *
 * @param  {object} runInfo The new info to render
 *
 * @return {object}    An action object with a type of DISPLAY_CURR_DOGS
 */
export function displayCurrDogs(runInfo) {
  return {
    type: DISPLAY_CURR_DOGS,
    runInfo,
  }
}

/**
 * Fires off saga watch for this action type
 */
export function runInfoSagaWatcher() {
  return {
    type: RUN_CURR_DOGS_SAGA,
  }
}
