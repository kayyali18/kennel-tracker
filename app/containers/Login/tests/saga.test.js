/**
 * Tests for LoginPage sagas
 */

import { put, takeLatest } from 'redux-saga/effects'

import { RUN_TOKEN_SAGA } from '../constants'
import runTokenSaga, { getToken } from '../saga'
import { clearUserCredentials } from '../actions'

describe('getToken Saga', () => {
  let getTokenGenerator

  beforeAll(() => {
    getTokenGenerator = getToken()

    // describe the selector (makeSelectEmail)
    const selectDescriptor = getTokenGenerator.next().value
    expect(selectDescriptor).toMatchSnapshot()

    const selectDescriptor2 = getTokenGenerator.next().value
    expect(selectDescriptor2).toMatchSnapshot()
  })

  it('should dispatch the clearUserCredentials action', () => {
    const putDescriptor = getTokenGenerator.next().value
    expect(putDescriptor).toEqual(put(clearUserCredentials()))
  })
})

describe('runTokenSaga', () => {
  const runTokenSagaTest = runTokenSaga()

  it('should start a task to watch for RUN_TOKEN_SAGA', () => {
    const takeLatestDescriptor = runTokenSagaTest.next().value
    expect(takeLatestDescriptor).toEqual(takeLatest(RUN_TOKEN_SAGA, getToken))
  })
})
