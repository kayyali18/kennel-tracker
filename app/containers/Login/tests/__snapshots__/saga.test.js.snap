// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[` 2`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;
